### PR TITLE
feature/45507 Azure key rotation

### PR DESCRIPTION
**Pull request by GuyHarwood**
create key rotation scripts for...

- [x] redis
- [x] service-bus
- [x] storage account
- [x] add integration test for entire process

### CODE REVIEW

**Review by jon-shipley — APPROVED** [review]
LGTM 👍

File: deploy/azure/key-vault/key-rotation/integration-test.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+then
+  fail "Storage Account Primary Connection String in key vault does not match actual value"
+fi
+
+if [[ $SB_CON_STRING_KV != $SB_CON_STRING_ACTUAL ]];
+then
+  fail "Service Bus Connection String for $SERVICE_BUS_USER in key vault does not match actual value"
+fi
+
+echo "primary keys test successful."
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply deploy/azure/key-vault/key-rotation/integration-test.sh:67]: good call, its a bit unwieldly

**Comment by GuyHarwood** [comment]
@jon-shipley all changes addressed

File: deploy/azure/key-vault/key-rotation/integration-test.sh [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+function get_key_vault_values()
+{
+  echo "extracting current key vault values for comparison..."
+  REDIS_KEY_KV=$(az keyvault secret show --name "RedisKey" --vault-name $KEY_VAULT_NAME | jq -r .value)
+  STORAGE_KEY_KV=$(az keyvault secret show --name "StorageAccountKey" --vault-name $KEY_VAULT_NAME | jq -r .value)
+  STORAGE_CON_STRING_KV=$(az keyvault secret show --name "StorageAccountConnectionString" --vault-name $KEY_VAULT_NAME | jq -r .value)
+  SB_CON_STRING_KV=$(az keyvault secret show --name "ServiceBusConnectionString-$SERVICE_BUS_USER" --vault-name $KEY_VAULT_NAME | jq -r .value)
+
```
Comment (jon-shipley) [inline deploy/azure/key-vault/key-rotation/integration-test.sh:42]: extra line

File: deploy/azure/key-vault/key-rotation/service-bus.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+SERVICE_BUS_NAME=$3 # service bus instance
+KEY_TYPE=$4 # accepted values are 'primary' or 'secondary'
+SERVICE_BUS_USER=$5 # the target user for key rotation
+
+SERVICE_BUS_KEY_TYPE="" # valid values are 'PrimaryKey', 'SecondaryKey'
+KEY_IDENTIFIER="" # valid valuees are 'primaryConnectionString', 'secondaryConnectionString'
+
+# Azure CLI uses slightly different identifiers for key type across different services :-/
+# align them for each service...
+if [ "$KEY_TYPE" = "primary" ]
```
Comment (jon-shipley) [inline deploy/azure/key-vault/key-rotation/service-bus.sh:20]: There is a deviation from the previous standard where the `if` statement was terminated by a semi-colon.  I think it is better if we maintain consistency. 

Having the semi-colon seems like the more robust code to me. As it allows the `then` keyword to be on the same line as the `if`, which is a common format.

File: deploy/azure/key-vault/key-rotation/storage.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+RES_GROUP=$1 # target resource group
+KEY_VAULT_NAME=$2 # key vault instance
+STORAGE_ACCOUNT_NAME=$3 # storage account
+KEY_TYPE=$4 # accepted values are 'primary' or 'secondary'
+
+STORAGE_ACCOUNT_KEY_TYPE="" # valid values are 'primary', 'secondary'
+
+# Azure CLI uses slightly different identifiers for key type across different services :-/
+# align them for each service...
+if [ "$KEY_TYPE" = "primary" ]
```
Comment (jon-shipley) [inline deploy/azure/key-vault/key-rotation/storage.sh:18]: Suggest we use the semi-colons in this file too.

**Review by jon-shipley — APPROVED** [review]
LGTM 💯 

Just some style comments.  Up to you.

**Pull request merged**